corehq/apps/domain/tests/test_domain_name_generation.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import print_function, unicode_literals
+
+from django.test import TestCase
+
+from corehq.apps.domain.models import Domain
+
+
+class DomainNameGenerationTestBase(TestCase):
+    def setUp(self):
+        self.domains = []
+
+    def add_domain(self, name):
+        domain = Domain(name=name)
+        domain.save()
+        self.domains.append(domain)
+
+    def tearDown(self):
+        for domain in self.domains:
+            domain.delete()
+
+
+class DomainNameGenerationBasicTest(DomainNameGenerationTestBase):
```

Comment:
nice - I think you can split out the tests that are specifically exercising the different edge cases and 

1) for anything that doesn't have to depend on the DB pull out those tests and use `SimpleTestCase` so they run faster
2) for anything that does depend on the DB you can leave it (or if you're feeling especially motivated you can use mocks)